examples/ffva/src/dfu_i2c/dfu_servicer.c

```diff
@@ -0,0 +1,272 @@
+    }
+
+    if(appconfI2C_CTRL_ENABLED > 0)
+    {
+        control_ret_t dc_ret;
+        debug_printf("Calling device_control_servicer_register(), servicer ID %d, on tile %d, core %d.\n", servicer->id, THIS_XCORE_TILE, rtos_core_id_get());
+
+        dc_ret = device_control_servicer_register(&servicer_ctx,
+                                            device_control_ctxs,
+                                            1,
+                                            resources, servicer->num_resources);
+        debug_printf("Out of device_control_servicer_register(), servicer ID %d, on tile %d. servicer_ctx address = 0x%x\n", servicer->id, THIS_XCORE_TILE, &servicer_ctx);
+    }
+
+ #if appconfI2C_CTRL_ENABLED && ON_TILE(I2C_CTRL_TILE_NO)
+    // Start I2C slave.
+    // This ends up calling device_control_resources_register() which has a strange non-yielding implementation,
+    // where it waits for servicers to register with the device control context. That's why, control_start_io_tasks()
+    // is not called from platform_start(). Additionally, if there is only one xcore core dedicated for all RTOS tasks,
+    // this design will not work, since device_control_resources_register() will not yield and the servicers wouldn't get
+    // scheduled so they could register with the device control leading to an eventual timeout error from device_control_resources_register().
+
+    rtos_i2c_slave_start(i2c_slave_ctx,
+                         device_control_i2c_ctx,
+                         (rtos_i2c_slave_start_cb_t) device_control_i2c_start_cb,
+                         (rtos_i2c_slave_rx_cb_t) device_control_i2c_rx_cb,
+                         (rtos_i2c_slave_tx_start_cb_t) device_control_i2c_tx_start_cb,
+                         (rtos_i2c_slave_tx_done_cb_t) NULL,
+                         NULL,
+                         NULL,
+                         appconfI2C_INTERRUPT_CORE,
+                         appconfI2C_TASK_PRIORITY);
+#endif
+    vPortFree(resources);
+
+    // The first call to device_control_servicer_cmd_recv triggers the device_control_i2c_start_cb() through I2S slave ISR call somehow. device_control_i2c_start_cb() and device_control_servicer_cmd_recv() have to be in separate logical cores
+    // and be scheduled simultaneously for this to work.
+    if(APP_CONTROL_TRANSPORT_COUNT > 0)
+    {
+        for(;;){
+            device_control_servicer_cmd_recv(&servicer_ctx, read_cmd, write_cmd, servicer, RTOS_OSAL_WAIT_FOREVER);
+        }
+    }
+    else
+    {
+        for(;;){
+            vTaskDelay(pdMS_TO_TICKS(100));
+        }
+    }
+    // TODO: Enable lines below when the rest of DFU code is included
+    /*xTaskCreate(
+        dfu_int_state_machine,
+        "DFU state machine task",
+        RTOS_THREAD_STACK_SIZE(dfu_int_state_machine),
+        NULL,
+        uxTaskPriorityGet(NULL), // Same priority so should run after this task
+        NULL
+    );*/
+    
+    if(appconfI2C_CTRL_ENABLED > 0)
```

Comment:
If the DFU servicer task is created only when appconfI2C_CTRL_ENABLED is true, then a check here is not needed.